app/models/katello/provider.rb

```diff
@@ -33,6 +34,12 @@ class Provider < Katello::Model
     scope :custom, -> { where(:provider_type => CUSTOM) }
     scope :anonymous, -> { where(:provider_type => ANONYMOUS) }
 
+    def only_cdn_https
+      if repository_url.present? && repository_url.downcase.start_with?('https') && !repository_url.downcase.include?('cdn.redhat.com')
```

Comment:
Would it be more accurate to use do an exact match against "https://cdn.redhat.com" rather than `include?`? Just in case some users has 'cdn.redhat.com' in their url. (hopefully they shouldn't anyway)